File: Manual Copies/112-Path-Sum/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
  public boolean hasPathSum(TreeNode root, int targetSum) {
      // If no root node, there is no path
      if (root == null) return false;
      // If leaf
      else if (root.left == null && root.right == null) {
          // Check if targetSum is met (sum is updated with each recursive call)
          return targetSum == root.val;
      }
      // If not leaf, recursively call both left and right nodes checking each path
      return hasPathSum(root.left, targetSum - root.val) || hasPathSum(root.right, targetSum - root.val);
  }
}

File: Manual Copies/112-Path-Sum/Solution.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} targetSum
 * @return {boolean}
 */
var hasPathSum = function(root, targetSum) {
    // Base case 1: If the root is null, return false since there is no path
    if (root === null) {
        return false;
    }
    // If the root is a leaf node, check if the value of the root is equal to the target sum
    // NOTE: Target sum is updated to be the target sum minus the value of the node in the previous recursive call
    else if (root.left === null && root.right === null) {
        return targetSum === root.val;
    }
    // If the root is not a leaf node, check if the target sum minus the value of the root is equal to the sum of the left subtree or the right subtree
    return hasPathSum(root.left, targetSum - root.val) || hasPathSum(root.right, targetSum - root.val);
};